qermit/zero_noise_extrapolation/zne.py

```diff
@@ -170,6 +170,71 @@ def gate(circ: Circuit, noise_scaling: float, _allow_approx_fold: bool) -> Circu
 
         return folded_c
 
+    def odd_gate(circ: Circuit, noise_scaling: float) -> Circuit:
+        """Noise scaling by gate folding. In this case odd gates :math:`G` are
+        replaced :math:`GG^{-1}G` until the number of gates is sufficiently
+        scaled.
+
+        :param circ: Original circuit to be folded.
+        :type circ: Circuit
+        :param noise_scaling: Factor by which to increase the noise.
+        :type noise_scaling: float
+        :return: Folded circuit implementing identical unitary to the initial circuit.
+        :rtype: Circuit
+        """
+
+        c_dict = circ.to_dict()
+
+        fold = True
+        folded_command_list = []
+
+        for command in c_dict["commands"]:
+
+            if fold:
+                # for _ in range(noise_scaling - 1):
+                command_circ_dict = c_dict.copy()
```

Comment:
small optimisation suggestion

put this outside the loop:
```
command_circ_dict= {key: val for key, val in c_dict.items() if key != "commands"}
```
then in the loop

```
command_circ_dict.update({"commands": [command]})
```

Reply:
Nice, thanks!